ha/setup/ha_setup.py

```diff
@@ -194,30 +196,79 @@ def process(self):
         cluster_secret = Cipher.decrypt(key, cluster_secret.encode('ascii')).decode()
         s3_instances = self._get_s3_instance(machine_id)
 
-        try:
-            # Create cluster
-            Log.info(f"Creating cluster: {cluster_name} with node: {node_name}")
-            cluster_output: str = self._cluster_manager.cluster_controller.create_cluster(
-                cluster_name, cluster_user, cluster_secret, node_name)
-            Log.info(f"Cluster creation output: {cluster_output}")
-            # TODO: Handle race condition cluster and resource create with global check.
-            if json.loads(cluster_output).get("status") == STATUSES.SUCCEEDED.value:
-                # Put cluster in standby mode
-                standby_output: str = self._cluster_manager.node_controller.standby(node_name)
-                Log.info(f"Put node in standby output: {standby_output}")
-                if json.loads(standby_output).get("status") != STATUSES.FAILED.value:
-                    Log.info("Creating pacemaker resources")
-                    create_all_resources(s3_instances=s3_instances)
-                    Log.info("Created pacemaker resources successfully")
+        # Check if the cluster exists already, if not, create one.
+        cluster_exists = self._confstore.key_exists(const.CLUSTER_CONFSTORE_NODES_KEY)
+        Log.info(f"Cluster exists? {cluster_exists}")
+        if cluster_exists == False:
+            try:
+                # Create cluster
+                Log.info(f"Creating cluster: {cluster_name} with node: {node_name}")
+                cluster_output: str = self._cluster_manager.cluster_controller.create_cluster(
+                    cluster_name, cluster_user, cluster_secret, node_name)
+                Log.info(f"Cluster creation output: {cluster_output}")
+                # TODO: Handle race condition cluster and resource create with global check.
```

Comment:
which race condition?

Reply:
This is an existing comment, @ajay-paratmandali can you please provide more info?

Reply:
It was rase condition between

create_cluster()
create_resource()
But it got addressed by 

`self._confstore.set(f"{const.CLUSTER_CONFSTORE_NODES_KEY}/{node_name}")`
we can remove this TODO

Reply:
Removed the TODO.